### PR TITLE
Es gstudio api

### PR DESCRIPTION
**Pull request by siddhudhangar**
**ES Completed Tasks as below:**
1. In E-library, Filters , pagination , file statistics , file listing and counts of all listed files such as images,audios,videos ...  etc.
2. In E-book, Filters, pagination and file listing.
3. In Search text box, Searching now happening through ES.

All above features are working with help of ES.

### CODE REVIEW

File: doc/deployer/es_initialize.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+from gnowsys_ndf.settings import GSTUDIO_SITE_NAME
+from gnowsys_ndf.local_settings import GSTUDIO_DOCUMENT_MAPPING
+
+
+##### use the below commented lines if you are working with Python 2.x   #####
+# reload(sys)  
+# sys.setdefaultencoding('UTF8')
+
+es = Elasticsearch("http://elastic:changeme@gsearch:9200", timeout=100, retry_on_timeout=True)
```
Comment (kedar2a) [inline doc/deployer/es_initialize.py]: **Get `username` and `password` from `settings.py` file.**

Sensitive information should not be there in code files and in git. Best practise is to get it from `local_settings.py` or environmental variables.

**Comment by kedar2a** [comment]
Do `git rm` for following files:
- `ATs.csv`
- `RTs.csv`
- `STs_run1.csv`
- `STs_run1.csv`
and their respective JSON files

**Comment by siddhudhangar** [comment]
i want check dependency of result_detailed_view.html template.... so that right now i kept that file as it is.

File: gnowsys-ndf/gnowsys_ndf/ndf/templates/ndf/add_asset.html [hunk 2]
```diff
@@ -148,14 +151,14 @@ <h3 class="left overlay-title">&nbsp;
             <label>
 
                     {% if title == "raw material"  or 'raw@material' in asset_obj.tags %}
-                        {% trans "Resource Description" %}
+                        {% trans "Resource Description" %} <h9 style="color: red;" id="req1"></h9>
 
                     {% elif title == "gallery"  or 'asset@gallery' in asset_obj.tags %}
-                         {% trans "Folder Description" %}
+                         {% trans "Folder Description" %}  <h9 style="color: red;" id="req1"></h9>
                     {% else %}
```
Comment (kedar2a) [inline gnowsys-ndf/gnowsys_ndf/ndf/templates/ndf/add_asset.html]: check for uniquness of html element `id`

File: gnowsys-ndf/gnowsys_ndf/ndf/templates/ndf/card_gsearch.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<!-- {% load i18n %}
+{% load ndf_tags %}
```
Comment (kedar2a) [inline gnowsys-ndf/gnowsys_ndf/ndf/templates/ndf/card_gsearch.html]: Uncomment this file content

**Comment by gnowgi** [comment]
code being used in beta, and in the deployment.  closing this request now for the master.  can be reissued when needed.

**Pull request closed without merging**